ckanext/dataset_subscriptions/plugin.py

```diff
@@ -1,14 +1,91 @@
 import ckan.plugins as plugins
 import ckan.plugins.toolkit as toolkit
+import ckan.logic as logic
+import ckan.lib.helpers as helpers
+import ckan.lib.email_notifications as email_notifications
+import ckan.model as model
+import ckan.lib.base as base
+from ckan.common import ungettext, config
+
+
+@toolkit.chained_action
+@toolkit.side_effect_free
+def send_email_notifications(original_action, context, data_dict):
+    email_notifications._notifications_functions = [dms_notification_provider]
+    return original_action(context, data_dict)
+
+
+def dms_notification_provider(user_dict, since):
+    context = {'model': model, 'session': model.Session,
+               'user': user_dict['id']}
+    activity_list = logic.get_action('dashboard_activity_list')(context, {})
+    # Return only activites that contain dataset changes
+    activity_list = [activity for activity in activity_list
+                     if activity['user_id'] != user_dict['id']
+                     and 'package' in activity['activity_type']]
+    # We want a notification per changed dataset, not a list of all changes
+    deduplicated_activity_list = list({item["object_id"]: item for item in activity_list}.values())
+    # Add dataset name to activity list
+    for index, activity in enumerate(deduplicated_activity_list):
+        object_id = activity['object_id']
+        dataset = toolkit.get_action('package_show')(
+                    context, {'id': object_id})
+        dataset_name = dataset['name']
+        print(dataset_name)
+        deduplicated_activity_list[index]['dataset_name'] = dataset_name
```

Comment:
create a helper function to enhance activities detail with dataset names. You can name this function nicely so the line 28 comment is not needed. e.g.
```
def _add_dataset_name_to_activities(activity_list):
   for index, activity in activity_list:
      [...]
```